Week 1 [Step 3] jinyoo

Rob-Yoo opened the pull request:
@Kioding
Step3 코드 구현 및 순서도 작성한 것 제출합니다!

Commit:
Feat: 주문에 따른 빼뺴로 그리기 구현

- drawStick: 길이에 따른 막대 그리는 함수
- drawBody: 몸통, 토핑을 매개변수로 받아 각각을 그리는 함수
- drawFullBody: 길이에 따라 몸통을 그리는 함수
- drawBBR: 각종 매개변수를 받아 위의 세 메서드를 호출하는 함수

Commit:
Chore: Step2, Step3 디렉토리 생성

- Step2, Step3 디렉토리로 나누어서 파일들 관리

Commit:
Chore : Step3 순서도 작성
docs : README.md 수정

- Step3 순서도 작성
- README.md의 이미지 파일들 경로 수정 및 Step3 순서도 첨부

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,52 @@
+    var topping: String = orderedTopping
+    if (topping.isEmpty){
+       topping = " "
+    }
```
swift 에서는 소괄호가 없어도 조건문이 가능합니다.
연산 방식이 많아 가독성을 위한 소괄호가 아니라면 지워도 괜찮을 것 같네요 :)
다른 부분도 소괄호가 있다면 같은 코멘트입니다!

Kioding replied:
isEmpty 를 사용하신 점이 포인트네요!

Rob-Yoo replied:
앗,, 습관적으로 소괄호를 써버렸네요...주의하겠습니다!!

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,52 @@
+    }
+}
+
+func drawBBR(bodyLength: Int, body: String, orderedTopping: String, stickLength: Int) {
```
`drawBBR()` 에서 BBR은 무슨 뜻일까요?
swift 에서는 축약어를 지양하고 있습니다.
각 함수는 함수명만으로 무슨 역할을 하는지 나타낼 수 있다면 어떤 이름을 붙이면 좋을까요? 😆

Rob-Yoo replied:
아하! 축약어를 지양하고 있군요!! 구글에 빼빼로를 영어로 어떻게 쓰나 봤더니 pepero라고 하더라고요. drawPepero라고 다시 네이밍 하겠습니다!

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,52 @@
+    print("길이: \(bodyLength)")
+    print("몸통: \(body)")
+    print("토핑: \(orderedTopping)")
+    print("막대길이: \(stickLength)\n")
```
print를 여러줄 한다면 multi-line print 를 사용할 수 있습니다.
어떻게 사용하는 지 검색해보고 적용해보면 좋을 것 같아요 😃

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,52 @@
+    print(topping + body + topping)
+}
+
+func drawFullBody(body: String, toppping: String, bodyLength: Int) {
```
`toppping` 은 오타로 보이네요 😁

Rob-Yoo replied:
앗,,, 이럴수가,,,,

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,52 @@
+    for _ in 1...stickLength {
+            print(" ", " ", separator: "\(stick)")
+    }
+}
```
stickLength 만큼 for문을 돌린다는 점을 확인할 수 있어 좋네요!

Commit:
Refact : 함수, 변수 네이밍 수정 및 if 문 수정

- drawBBR를 drawPepero로 수정함
- drawFullBody 함수에서 toppping 매개변수를 topping으로 수정함
- if문의 소괄호를 없앰

Commit:
Refact : drawPepero 함수에서 multi-line string 사용

- 기존의 4번의 print함수를 사용한 것에서 multi-line string을 이용해서 한번의 print함수를 사용한 것으로 변경

Commit:
Chore : 순서도 수정

- 순서도 파일 이름 변경
- 순서도의 각 빈칸들의 내용을 문법적인 요소가 안들어가게 수정

Commit:
Chore : 루트 디렉토리에 FlowCharts 폴더 생성

- FlowCharts 폴더를 생성
- 안에 Step2라는 폴더를 생성하고 Step2의 순서도 관련 파일들 보관

Commit:
Chore : Step2의 순서도 관련 파일들 모두 옮김

Commit:
Chore : Step3의 순서도 관련 파일들 이동

- 프로젝트 디렉토리에서 루트 디렉토리의 FlowCharts/Step3로 이동

Commit:
Chore: README.md 수정

Commit:
Chore: README에서 파일 경로 오타 수정

Commit:
Chore: README에서 파일 경로 오타 수정

Commit:
Chore : Xcode 상에서 디렉토리 생성 및 파일 관리

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,54 @@
+}
+
+func drawFullBody(body: String, topping: String, bodyLength: Int) {
+    if body.hasPrefix("|") {
```
확장성을 생각한다면 `|0|` 의 경우에 if문을 타는 것도 좋을 것 같네요 :)

Rob-Yoo replied:
@Kioding
|0| 의 경우에 if문을 타는 것이 좋다는 것이 무슨 말씀인지 잘 모르겠습니다ㅠㅠㅠ

Kioding replied:
아래 코드의 경우는 빼빼로 바디가 `|0|`의 경우 설계한대로 잘 타지만
`|||`, `|12`, `|as` 가 들어와도 타는 구조겠죠?!
```swift
 if body.hasPrefix("|")
```

<br>

그래서 만약 다른 빼빼로 바디가 들어올 때는 적용하고 싶지 않다면 
아래의 방법도 있다 라는 의견이었습니다 😆
```swift
if body == "|0|" 
```

<br>

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,54 @@
+        }
+        if bodyLength % 2 == 1 {
+            drawBody(body: "| |", topping: topping)
+        }
```
호오 많이 고민한 흔적이 보이는 구간이네요!
`halfLength` 를 이용하여 마지막을 홀수일 경우만 처리하셨네요 👍 
짝수 - 홀수일 경우에 따라 처리할 수도 있을 것 같네요! 
> (정답은 없고 이러한 방법도 있다의 멘트입니다!)

Kioding commented on CodeStarterCamp_Week1/Step3/main.swift:
```diff
@@ -0,0 +1,54 @@
+    몸통: \(body)
+    토핑: \(orderedTopping)
+    막대길이: \(stickLength)\n
+    """)
```
멀티라인으로 바꿔주셨네요! 
지난 번 하드코딩에서 매개변수를 받아 오니 일일히 적어주지 않아도 되는 점에서 훨씬 좋아졌습니다 👏👏👏
print 이후는 다른 함수만 호출하는 것처럼 print 만 하는 부분을 함수로 만들어 주어 호출해주면 
`drawPepero()` 에서 3개의 함수를 호출하는 구나 라며 순서의 흐름을 보기 쉬울 것 같네요!

Commit:
Refactor : 실행되지 않는 소스코드의 함수 실행 코드 삭제

Commit:
Refactor : 함수 네이밍 수정

Commit:
Refactor : 함수 네이밍 재수정 및 주문 정보 출력하는 함수 새로 추가

- makeOrderedPepero에서 handleOrderedPepero로 함수 네이밍 재수정
- 주문 정보를 출력하는 부분을 기존의 print함수를 바로 호출하는 방식에서 printOrderedInformation라는 함수를 만들어 호출하는 방식으로 가독성을 높힘
